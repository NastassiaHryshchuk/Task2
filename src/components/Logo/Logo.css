.logo {
    text-decoration: none;
    color: rgb(247, 66, 96);
}