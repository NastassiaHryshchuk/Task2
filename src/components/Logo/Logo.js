import React from 'react';
import classes from './Logo.css';

const Logo = () => {
  return <a href="#" className={classes.logo}>netflixroulette</a>
};

export default Logo;
