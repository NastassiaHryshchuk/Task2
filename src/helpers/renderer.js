import React from 'react';
import { renderToString } from 'react-dom/server';
import { StaticRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import Routes from '../Routes/Routes';

export default(req, store) => {
  const content = renderToString(
    <Provider store={store}>
      <StaticRouter location={req.path} context={{}}>
        <Routes />
      </StaticRouter>
    </Provider>,
  );

  return `
    <!DOCTYPE html>
    <html>
      <head>
        <title>SSR</title>
      </head>
      <body>
        <div id="app">${content}</div>
        <script>
          window.INITIAL_STATE = ${JSON.stringify(store.getState())}
        </script>
        <script src="bundle.js"></script>
      </body>
    </html>
  `;
};
