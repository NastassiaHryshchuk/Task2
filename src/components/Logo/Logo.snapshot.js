import React from 'react';
import renderer from 'react-test-renderer';

import Logo from './Logo';

describe('Logo Snapshot', () => {
  test('renders', () => {
    const component = renderer.create(
      <Logo />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

