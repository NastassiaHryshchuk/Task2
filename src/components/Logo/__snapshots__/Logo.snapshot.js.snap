// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo Snapshot renders 1`] = `
<a
  className="logo"
  href="#"
>
  netflixroulette
</a>
`;
